python/ray/ml/train/integrations/sklearn/sklearn_trainer.py

```diff
@@ -0,0 +1,133 @@
+from typing import Union, Callable, Dict, Iterable, Optional
+
+import numpy as np
+import pandas as pd
+from ray.ml.trainer import GenDataset
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.preprocessor import Preprocessor
+from ray.util import PublicAPI
+from ray.ml.trainer import Trainer
+from ray.ml.checkpoint import Checkpoint
+
+from sklearn.base import BaseEstimator
+from sklearn.model_selection import BaseCrossValidator
+
+# some nice properties of Sklearn models:
+# - they are always pickleable
+# - no logic is present in __init__
+
+ArrayType = Union[pd.DataFrame, np.ndarray]
+MetricType = Union[str, Callable[[BaseEstimator, ArrayType, ArrayType], float]]
+ScoringType = Union[MetricType, Iterable[MetricType], Dict[str, MetricType]]
+CVType = Union[int, Iterable, BaseCrossValidator]
+
+
+@PublicAPI(stability="alpha")
+class SklearnTrainer(Trainer):
+    """A Trainer for scikit-learn estimator training.
+
+    This Trainer runs the ``fit`` method of the given estimator in a
+    non-distributed manner on a single Ray Actor.
+
+    The ``n_jobs`` (or ``thread_count``) parameters will be set to match
+    the number of CPUs assigned to the Ray Actor.
+
+    Example:
+        .. code-block:: python
+
+            import ray
+
+            from ray.ml.train.integrations.sklearn import SklearnTrainer
+            from sklearn.ensemble import RandomForestRegressor
+
+            train_dataset = ray.data.from_items(
+                [{"x": x, "y": x + 1} for x in range(32)])
+            trainer = SklearnTrainer(
+                sklearn_estimator=RandomForestRegressor,
+                label_column="y",
+                scaling_config={
+                    "trainer_resources": {"CPU": 4}
```

Comment:
`trainer_resources` is not a valid key in the scaling config right?

Reply:
It should be? It's in the dataclass

Reply:
oops sorry, yes you're right- forgot that we had changed this!